sql/tests: TestRandomSyntaxGeneration failed [CREATE SCHEDULE ... FOR CHANGEFEED parsing/formatting]

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/12265129?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/12265129?buildTab=artifacts#/) on release-23.1 @ [ff444373b429926c6686eae90ad2fcc9551c4dff](https://github.com/cockroachdb/cockroach/commits/ff444373b429926c6686eae90ad2fcc9551c4dff):


```
    rsg_test.go:870: 2m25s of 5m0s: 352977 executions, 18219 successful
    rsg_test.go:870: 2m30s of 5m0s: 365243 executions, 18790 successful
    rsg_test.go:870: 2m35s of 5m0s: 378871 executions, 19419 successful
    rsg_test.go:870: 2m40s of 5m0s: 391989 executions, 20012 successful
    rsg_test.go:870: 2m45s of 5m0s: 404467 executions, 20640 successful
    rsg_test.go:870: 2m50s of 5m0s: 417049 executions, 21258 successful
    rsg_test.go:870: 2m55s of 5m0s: 430190 executions, 21908 successful
    rsg_test.go:870: 3m0s of 5m0s: 443737 executions, 22556 successful
    rsg_test.go:870: 3m5s of 5m0s: 457064 executions, 23187 successful
    rsg_test.go:870: 3m10s of 5m0s: 468898 executions, 23753 successful
    rsg_test.go:870: 3m15s of 5m0s: 481767 executions, 24350 successful
    rsg_test.go:870: 3m20s of 5m0s: 494787 executions, 24936 successful
    rsg_test.go:870: 3m25s of 5m0s: 508292 executions, 25538 successful
    rsg_test.go:870: 3m30s of 5m0s: 522070 executions, 26130 successful
    rsg_test.go:870: 3m35s of 5m0s: 534159 executions, 26692 successful
    rsg_test.go:870: 3m40s of 5m0s: 547356 executions, 27285 successful
    rsg_test.go:870: 3m45s of 5m0s: 560506 executions, 27910 successful
    rsg_test.go:870: 3m50s of 5m0s: 573694 executions, 28559 successful
    rsg_test.go:870: 3m55s of 5m0s: 587063 executions, 29219 successful
    rsg_test.go:870: 4m0s of 5m0s: 599986 executions, 29816 successful
    rsg_test.go:870: 4m5s of 5m0s: 612662 executions, 30440 successful
    rsg_test.go:870: 4m10s of 5m0s: 625368 executions, 31042 successful
    rsg_test.go:870: 4m15s of 5m0s: 638332 executions, 31644 successful
    rsg_test.go:870: 4m20s of 5m0s: 651392 executions, 32195 successful
    rsg_test.go:870: 4m25s of 5m0s: 664964 executions, 32780 successful
    rsg_test.go:870: 4m30s of 5m0s: 678740 executions, 33345 successful
    rsg_test.go:870: 4m35s of 5m0s: 691352 executions, 33965 successful
    rsg_test.go:870: 4m40s of 5m0s: 705347 executions, 34588 successful
    rsg_test.go:870: 4m45s of 5m0s: 718555 executions, 35188 successful
    rsg_test.go:870: 4m50s of 5m0s: 732122 executions, 35832 successful
    rsg_test.go:870: 4m55s of 5m0s: 744263 executions, 36399 successful
    rsg_test.go:212: Context cancelled while executing: "DROP OWNED BY GENERATED , UNLOGGED , ENUM RESTRICT"
    rsg_test.go:212: Context cancelled while executing: "CREATE DATABASE IF NOT EXISTS ident"
    rsg_test.go:212: Context cancelled while executing: "CREATE DATABASE IF NOT EXISTS ident"
    rsg_test.go:212: Context cancelled while executing: "COMMENT ON CONSTRAINT MULTIPOINTZM ON FAMILY . FAMILY . ISNULL IS 'string'"
    rsg_test.go:212: Context cancelled while executing: "CREATE DATABASE IF NOT EXISTS ident"
    rsg_test.go:212: Context cancelled while executing: "CREATE DATABASE IF NOT EXISTS ident"
    rsg_test.go:212: Context cancelled while executing: "UNLISTEN NOTHING_AFTER_RETURNING . EXISTS . ident"
    rsg_test.go:212: Context cancelled while executing: "CREATE DATABASE IF NOT EXISTS ident"
    rsg_test.go:212: Context cancelled while executing: "COPY NOTHING_AFTER_RETURNING FROM STDIN WITH ( DESTINATION 'string' )"
    rsg_test.go:212: Context cancelled while executing: "CREATE DATABASE IF NOT EXISTS ident"
    rsg_test.go:212: Context cancelled while executing: "IMPORT ident ( TRACING ) WITH JOBS"
    rsg_test.go:212: Context cancelled while executing: "COPY FAMILY . POSITION . ident ( ident , DEC ) FROM error"
    rsg_test.go:212: Context cancelled while executing: "COPY START . INDEX_BEFORE_PAREN TO error"
    rsg_test.go:212: Context cancelled while executing: "PREPARE MULTILINESTRINGZ ( POINT error ARRAY ) AS OPT PLAN 'string'"
    rsg_test.go:906: 757379 executions, 36982 successful
    rsg_test.go:914: cannot parse output of Format: sql="CREATE SCHEDULE PLACEHOLDER FOR CHANGEFEED ident . IS FAMILY ident INTO 'string' WITH OPTIONS ( BUCKET_COUNT , ident , CLOSE , ident , 'string' , INOUT ) RECURRING 'string'", formattedSQL="CREATE SCHEDULE 'placeholder' FOR CHANGEFEED TABLE ident.is FAMILY ident INTO 'string' WITH bucket_count, ident, close, ident, \"string\", \"inout\" RECURRING 'string'": at or near "with": syntax error
    rsg_test.go:317: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxGeneration825072116
--- FAIL: TestRandomSyntaxGeneration (301.91s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #112512 sql/tests: TestRandomSyntaxGeneration failed [tenant replication options] [C-test-failure O-robot T-multitenant branch-release-23.2 release-blocker]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32569

--- a/pkg/sql/parser/testdata/create_schedule
+++ b/pkg/sql/parser/testdata/create_schedule
@@ -1,3 +1,5 @@
+# Scheduled Backups Test
+
 parse
 CREATE SCHEDULE FOR BACKUP TABLE foo INTO 'bar' RECURRING '@hourly'
 ----
@@ -61,3 +63,13 @@ CREATE SCHEDULE IF NOT EXISTS 'baz' FOR BACKUP INTO 'bar' WITH revision_history 
 CREATE SCHEDULE IF NOT EXISTS ('baz') FOR BACKUP INTO ('bar') WITH revision_history = (true) RECURRING ('@daily') FULL BACKUP ('@weekly') WITH SCHEDULE OPTIONS first_run = ('now') -- fully parenthesized
 CREATE SCHEDULE IF NOT EXISTS '_' FOR BACKUP INTO '_' WITH revision_history = _ RECURRING '_' FULL BACKUP '_' WITH SCHEDULE OPTIONS first_run = '_' -- literals removed
 CREATE SCHEDULE IF NOT EXISTS 'baz' FOR BACKUP INTO 'bar' WITH revision_history = true RECURRING '@daily' FULL BACKUP '@weekly' WITH SCHEDULE OPTIONS _ = 'now' -- identifiers removed
+
+# Scheduled Changefeed Tests
+
+parse
+CREATE SCHEDULE FOR CHANGEFEED d.public.foo INTO 'webhook-https://0/changefeed?AWS_SECRET_ACCESS_KEY=nevershown' WITH initial_scan='only' RECURRING '@hourly'
+----
+CREATE SCHEDULE FOR CHANGEFEED TABLE d.public.foo INTO 'webhook-https://0/changefeed?AWS_SECRET_ACCESS_KEY=nevershown' WITH OPTIONS (initial_scan = 'only' ) RECURRING '@hourly' -- normalized!
+CREATE SCHEDULE FOR CHANGEFEED TABLE (d.public.foo) INTO ('webhook-https://0/changefeed?AWS_SECRET_ACCESS_KEY=nevershown') WITH OPTIONS (initial_scan = ('only') ) RECURRING ('@hourly') -- fully parenthesized
+CREATE SCHEDULE FOR CHANGEFEED TABLE d.public.foo INTO '_' WITH OPTIONS (initial_scan = '_' ) RECURRING '_' -- literals removed
+CREATE SCHEDULE FOR CHANGEFEED TABLE _._._ INTO 'webhook-https://0/changefeed?AWS_SECRET_ACCESS_KEY=nevershown' WITH OPTIONS (_ = 'only' ) RECURRING '@hourly' -- identifiers removed

--- a/pkg/sql/sem/tree/schedule.go
+++ b/pkg/sql/sem/tree/schedule.go
@@ -130,8 +130,9 @@ func (node *ScheduledChangefeed) Format(ctx *FmtCtx) {
 	ctx.FormatNode(node.SinkURI)
 
 	if node.Options != nil {
-		ctx.WriteString(" WITH ")
+		ctx.WriteString(" WITH OPTIONS (")
 		ctx.FormatNode(&node.Options)
+		ctx.WriteString(" )")
 	}
 
 	if node.Select != nil {

